MonoMod.RuntimeDetour/HookGen/HookILCursor.cs

```diff
@@ -97,17 +92,17 @@ public HookILCursor Clone()
             Index = index;
         }
 
-        public void End() {
-            _LastLabel = null;
-        }
-
         public void MarkLabel(HookILLabel label) {
-            _LastLabel = label;
             label.Target = Next;
+            _insertAfterLabels = new [] { label };
         }
 
         public void MoveAfterLabel() {
-            _LastLabel = LabelNext;
+			_insertAfterLabels = HookIL._Labels.Where(l => l.Target == Next).ToArray();
```

Comment:
```cs
HookIL._Labels.Where(l => l.Target == Next)
```

This specifically shouldn't be necessary, as each `HookILCursor` just holds what `HookIL` context it belongs to and its `ID`.

Creating a new label checks if there is already an ID assigned pointing to the same target. Setting the target of one label with ID `N` "sets" the target of all labels with ID `N`.

Just wondering, did it not work properly?